envoy/network/connection.h

```diff
@@ -194,6 +194,7 @@ class Connection : public Event::DeferredDeletable,
   /**
    * @return the connection info provider backing this connection.
    */
+  virtual ConnectionInfoSetter& connectionInfoProvider() PURE;
```

Comment:
The only line of contact I could find from the cluster config in upstream_impl.cc down to the connection was the ConnectionInfoSetter. So I had to surface it here.